examples/101.03-bundle.json

```diff
@@ -0,0 +1,110 @@
+{
+    "credentials": {
+      "hostkey": {
+        "env": "HOST_KEY",
+        "path": "/etc/hostkey.txt"
+      }
+    },
+    "custom": {
+      "com.example.backup-preferences": {
+        "frequency": "daily"
+      },
+      "com.example.duffle-bag": {
+        "icon": "https://example.com/icon.png",
+        "iconType": "PNG"
+      },
+      "dependencies": [
+        {
+          "requires": {
+            "bundle": "azure/mysql",
+            "version": {
+              "prereleases": "true",
+              "range": "5.7.x"
+            }
+          }
+        }
+      ]
+    },
+    "definitions": {
+      "http_port": {
+        "default": 80,
+        "maximum": 10240,
+        "minimum": 10,
+        "type": "integer"
+      },
+      "port": {
+        "maximum": 65535,
+        "minimum": 1024,
+        "type": "integer"
+      },
+      "string": {
+        "type": "string"
+      },
+      "x509Certificate": {
+        "contentEncoding": "base64",
+        "contentMediaType": "application/x-x509-user-cert",
+        "type": "string",
+        "writeOnly": true
+      }
+    },
+    "description": "An example 'thin' helloworld Cloud-Native Application Bundle",
+    "images": {
+      "my-microservice": {
+        "contentDigest": "sha256:aaaaaaaaaaaa...",
+        "description": "my microservice",
+        "image": "technosophos/microservice:1.2.3"
+      }
+    },
+    "invocationImages": [
+      {
+        "contentDigest": "sha256:aaaaaaa...",
+        "image": "technosophos/helloworld:0.1.0",
+        "imageType": "docker"
+      }
+    ],
+    "maintainers": [
+      {
+        "email": "matt.butcher@microsoft.com",
+        "name": "Matt Butcher",
+        "url": "https://example.com"
+      }
+    ],
+    "name": "helloworld",
+    "outputs": {
+      "fields": {
+        "clientCert": {
+          "definition": "x509Certificate",
+          "path": "/cnab/app/outputs/clientCert"
+        },
+        "hostName": {
+          "applyTo": [
+            "install"
+          ],
+          "definition": "string",
+          "description": "the hostname produced installing the bundle",
+          "path": "/cnab/app/outputs/hostname"
+        },
+        "port": {
+          "definition": "port",
+          "path": "/cnab/app/outputs/port"
+        }
+      }
+    },
+    "parameters": {
+      "fields": {
+        "backend_port": {
+          "definition": "http_port",
+          "description": "The port that the back-end will listen on",
+          "destination": {
+            "env": "BACKEND_PORT"
+          }
+        }
+      }
+    },
+    "requiredExtensions" : {
+        "dependencies" : "cnab.io/specs/depenencies.schema.json"
```

Comment:
JSON Schema doesn't require that any validation tools need to make any network calls and this is just a reference. We wouldn't actually be able to pull these down in any sort of disconnected mode of operations either, so I'm not sure that HTTPS for security really gives anything? 

I think for most extensions there will be some required functionality that needs to be handled ahead of time, so I think they are kind of "registered" with the tooling provider at least and this should just be a "well known" extension schema?

Reply:
Seems reasonable, but I'm still not clear what "well known" really means. For example, suppose a company is implementing their own custom extension (for use only by the company's runtime and bundles -- no-one else knows are cares about the extension). Does the extension need to be "published" in order to be "well known" or can it be kept private?

Reply:
I think we can remove the well known part? I'm a private company building my own extension that I don't plan on publishing (sad :() but I'd still reference it here with some sort of URI like reference yes? 

So maybe just remove the "well known" part? That was really trying to address that a bundle author could reference a published (i.e. well known) extension (dependencies for example) and a tool could validate that it supports that extension because it knows the schema reference)

Reply:
It seems like the schema reference needs to be unique across extensions but doesn't need to refer to a JSON schema at all. If true, I find that confusing. I'm wondering what the difference in function is between a schema reference and the reverse DNS extension name.

Reply:
Following up on our slack conversation:

Proposal: "requiredExtensions" just becomes an array of "EXTENSION NAME" in custom.

Reply:
Agreed, thanks.